lib/executor/base_config.go

```diff
@@ -0,0 +1,141 @@
+/*
+ *
+ * k6 - a next-generation load testing tool
+ * Copyright (C) 2019 Load Impact
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package executor
+
+import (
+	"fmt"
+	"regexp"
+	"strings"
+	"time"
+
+	"github.com/loadimpact/k6/lib/types"
+	null "gopkg.in/guregu/null.v3"
+)
+
+var executorNameWhitelist = regexp.MustCompile(`^[0-9a-zA-Z_-]+$`) //nolint:gochecknoglobals
+const executorNameErr = "the executor name should contain only numbers, latin letters, underscores, and dashes"
+
+// BaseConfig contains the common config fields for all executors
+type BaseConfig struct {
+	Name         string             `json:"-"` // set via the JS object key
+	Type         string             `json:"type"`
+	StartTime    types.NullDuration `json:"startTime"`
+	GracefulStop types.NullDuration `json:"gracefulStop"`
+	Env          map[string]string  `json:"env"`
+	Exec         null.String        `json:"exec"` // function name, externally validated
```

Comment:
Minor: same comment as for `ExecutorConfig.GetExec`. Consider renaming this to something more descriptive like `ExecFunctionName` or `FunctionName`.

Reply:
Yeah, this will probably get done in a separate PR, either to `master` or to this branch if we delay merging it in `master`, so I'll comment out or delete this code for now and fix the PR description/create a new issue about it.

But you understand the idea correctly - to make supporting different scenarios easier, we want to allow different executors the option to execute other `export`-ed functions besides the `default` one. And even further into the future, after the new HTTP API, we may allow the specification of an HTTP request directly, without the need for a JS VM.

Reply:
Ah, actually I'm not sure anymore if deleting/commenting `env`/`exec` is a good idea, even if we implement them in a separate PR later. I'd have to delete quite a few things, including some validation. Given that we know we want these either in the same PR or immediately after, I don't see a lot of harm in leaving them as they are, just waiting for the implementation.

Reply:
Sure, if it's too much work to extract it now, let's leave it as is.